website/static/js/pointers.js

```diff
@@ -19,132 +19,238 @@ var SEARCH_MY_PROJECTS_SUBMIT_TEXT = 'Search my projects';
+        self.loadingResults(true);
+        self.fetchResults();
     },
     fetchResults: function() {
         var self = this;
         self.errorMsg('');
         self.searchWarningMsg('');
-
-        if (self.query()) {
-            self.results([]); // clears page for spinner
-            self.loadingResults(true); // enables spinner
-
-            osfHelpers.postJSON(
-                '/api/v1/search/node/', {
-                    query: self.query(),
-                    nodeId: nodeId,
-                    includePublic: self.includePublic(),
-                    page: self.pageToGet()
-                }
-            ).done(function(result) {
-                if (!result.nodes.length) {
-                    self.errorMsg('No results found.');
-                } else {
-                    result.nodes.forEach(function(each) {
-                        if (each.isRegistration) {
-                            each.dateRegistered = new osfHelpers.FormattableDate(each.dateRegistered);
+        self.results([]); // clears page for spinner
+        self.selection([]);
+        var query = '', myProjects = '', pageNum = self.pageToGet()+1;
+        if (self.query()){
+            query += 'filter[title]='+self.query()+'&';
+        }
+        if (!self.includePublic()){
+            myProjects = 'users/me/';
+        }
+        var url = osfHelpers.apiV2Url(myProjects+self.inputType()+'/', {query: query+'page='+pageNum+'&embed=contributors&page[size]=4'});
+        var request = osfHelpers.ajaxJSON(
+            'GET',
+            url,
+            {'isCors': true});
+        request.done(function(response) {
+            var nodes = response.data;
+            if (!nodes.length) {
+                self.errorMsg('No results found.');
+            }
+            else {
+                var count = nodes.length;
+                var url = osfHelpers.apiV2Url('nodes/'+nodeId+'/node_links/', {});
+                var request = osfHelpers.ajaxJSON(
+                    'GET',
+                    url,
+                    {'isCors': true});
+                request.done(function(nl_response) {
+                    nodes.forEach(function(each) {
+                        if (each.type === 'registrations') {
+                            each.dateRegistered = new osfHelpers.FormattableDate(each.attributes.date_registered);
                         } else {
-                            each.dateCreated = new osfHelpers.FormattableDate(each.dateCreated);
-                            each.dateModified = new osfHelpers.FormattableDate(each.dateModified);
+                            each.dateCreated = new osfHelpers.FormattableDate(each.attributes.date_created);
+                            each.dateModified = new osfHelpers.FormattableDate(each.attributes.date_modified);
+                        }
+                        // each.link = '';
+                        var i;
+                        for (i = 0; i < nl_response.data.length; i++) {
```

Comment:
I think looping over this once and then using indexOf for each node will be faster